src/main.cpp

```diff
@@ -90,20 +91,54 @@ namespace
 
     auto rule_exists(irods::default_re_ctx&, const std::string& _rule_name, bool& _exists) -> irods::error
     {
-        _exists = (_rule_name == "pep_api_mod_avu_metadata_pre");
+        _exists = (_rule_name == "pep_api_mod_avu_metadata_pre") || (_rule_name == "pep_api_atomic_apply_metadata_operations_pre");
         return SUCCESS();
     }
 
     auto list_rules(irods::default_re_ctx&, std::vector<std::string>& _rules) -> irods::error
     {
         _rules.push_back("pep_api_mod_avu_metadata_pre");
+        _rules.push_back("pep_api_atomic_apply_metadata_operations_pre");
         return SUCCESS();
     }
 
-    auto exec_rule(irods::default_re_ctx&,
-                   const std::string& _rule_name,
-                   std::list<boost::any>& _rule_arguments,
-                   irods::callback _effect_handler) -> irods::error
+    auto rule_logic(const std::string& _object_name, ruleExecInfo_t& rei) -> irods::error
+    {
+        const auto config = load_plugin_config(rei);
+        const auto& prefixes = config->at("prefixes");
+        if(std::any_of(prefixes.cbegin(), prefixes.cend(), [&_object_name](const json& prefix) { return boost::starts_with(_object_name, prefix.get_ref<const std::string&>()); })) {
+            if (config->count("admin_only") && config->at("admin_only").get<bool>()) {
```

Comment:
Prefer `.find` when you need to check for existence and then use the object.

Reply:
Where would the find be in this case?

Reply:
```cpp
const auto iter = config->find("admin_only");
```